woocommerce/woocommerce-ios
# [Engage] Universal links: Tracking

Closes: #7542 


### Description

On this PR we implement the tracking for the Universal Links feature. For the first iteration we want to track:
- Successful case with path as parameter
- Failure with whole URL as parameter

### Testing instructions

At the moment there is no easy way to test it as the domain verification is not yet implemented in woocommerce.com, so either you can wait for the domain verification to be added, or the next steps to test it now:

1. Add applinks:thelinkswootester.mystagingwebsite.com to the Associated Domains in the WooCommerce target screen. 
<img width="915" alt="Screenshot 2022-08-23 at 18 28 17" src="https://user-images.githubusercontent.com/1864060/186211830-d2a37a93-0e41-4082-8a76-92a6121ce8af.png">
This is my dummy pressable test site. (We do not have domain verification ready in woocommerce.com) I uploaded the [`apple-app-site-association`](https://thelinkswootester.mystagingwebsite.com/apple-app-site-association) with paths:

```
"paths":[
               "/payments",
	             "/orders/details"
            ]
```
2. Send yourself an email containing a link with the format 
https://thelinkswootester.mystagingwebsite.com/payments or 
https://thelinkswootester.mystagingwebsite.com/orders/details?blog_id=your storeID&order_id=an existing orderID Tap on it

3. Verify that the app opens and the events are tracked:
`🔵 Tracked universal_link_opened, properties: [AnyHashable("blog_id"): 205113380, AnyHashable("is_wpcom_store"): true, AnyHashable("path"): "/payments"]` or
`🔵 Tracked universal_link_opened, properties: [AnyHashable("blog_id"): 205113380, AnyHashable("path"): "/orders/details", AnyHashable("is_wpcom_store"): true]`

### Error case
1. Send your self an email containing a link with the wrong format 
2. https://thelinkswootester.mystagingwebsite.com/orders/details?blog_id=your storeID&**test_id**=an existing orderID
(This will fail because the parameter is test_id instead of order_id)
3. Verify that the app opens and bounces to Safari. The event is tracked:  `Tracked universal_link_failed, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("url"): https://thelinkswootester.mystagingwebsite.com/orders/details?blog_id=205293073&test_id=70, AnyHashable("blog_id"): 205113380]`


### Screenshots

N/A


---
